cmd/build.go

```diff
@@ -41,23 +41,28 @@ func Build(ctx context.Context) *cobra.Command {
 		Short: "Build (and optionally push) a Docker image",
 		RunE: func(cmd *cobra.Command, args []string) error {
 
+			if err := contextCMD.NewContextCommand().Run(ctx, &contextCMD.ContextOptions{}); err != nil {
+				return err
+			}
 			if okteto.IsOkteto() && options.Namespace != "" {
 				create, err := utils.ShouldCreateNamespace(ctx, options.Namespace)
 				if err != nil {
 					return err
 				}
 				if create {
-					err = namespace.ExecuteCreateNamespace(ctx, options.Namespace, nil)
+					nsCmd, err := namespace.NewNamespaceCommand()
 					if err != nil {
 						return err
 					}
+					nsCmd.CreateNamespace(ctx, &namespace.CreateOptions{Namespace: options.Namespace})
```

Comment:
same here... as is inside the namespace command maybe nsCmd.Create ? 
regarding Options, is there any other action options? EditOptions, DeleteOptions, could it be interesting to have an Options to use in any action?